nyxx.commander/lib/src/Commander.dart

```diff
@@ -94,22 +94,34 @@ class Commander with ICommandRegistrable {
     }
 
     // Find matching command with given message content
-    final messageContentParts = event.message.content.replaceFirst(prefix, "").trim().split(" ");
-    final matchingCommand = _commandEntities._findMatchingCommand(messageContentParts) as CommandHandler?;
+    final matchingCommand = _commandEntities._findMatchingCommand(event.message.content.replaceFirst(prefix, "").trim().split(" ")) as CommandHandler?;
 
     if(matchingCommand == null) {
       return;
     }
 
-    final fullCommandName = _getFullCommandFromMessage(matchingCommand, messageContentParts);
+    // Builds a RegEx that matches the full command including their parents and all possible
+    // aliases of the final command entity and their parents.
+    // Example: (?<finalCommand>(quote|q) (remove|rm))
+    // This will match the command `quote remove`, `q remove`, `quote rm` and `q rm`
+
+    final match = RegExp("(?<finalCommand>${matchingCommand.getFullCommandMatch()})").firstMatch(
+      event.message.content,
+    );
+
+    final finalCommand = match?.namedGroup("finalCommand");
+
+    if (finalCommand == null) {
+      return;
+    }
```

Comment:
Is that check really needed? Handler will stop executing higher I think we won't need that.

Reply:
In practice, finalCommand should never be null because a valid command was found using _findMatchingCommand, so the RegEx should also be able to figure out the full command.

I have removed the check.